radiation/python/radlw/rtrnmc_gt4py.py

```diff
@@ -0,0 +1,584 @@
+           NGB: Field[gtscript.IJ, (np.int32, (140,))],
+           htr: FIELD_FLT,
+           htrcl: FIELD_FLT,
+           htrb: Field[type_nbands],
+           totuflux: FIELD_FLT,
+           totdflux: FIELD_FLT,
+           totuclfl: FIELD_FLT,
+           totdclfl: FIELD_FLT,
+           clrurad: Field[type_nbands],
+           clrdrad: Field[type_nbands],
+           toturad: Field[type_nbands],
+           totdrad: Field[type_nbands],
+           gassrcu: Field[type_ngptlw],
+           totsrcu: Field[type_ngptlw],
+           trngas: Field[type_ngptlw],
+           efclrfr: Field[type_ngptlw],
+           rfdelp: FIELD_FLT,
+           fnet: FIELD_FLT,
+           fnetc: FIELD_FLT,
+           totsrcd: Field[type_ngptlw],
+           gassrcd: Field[type_ngptlw],
+           tblind: Field[type_ngptlw],
+           odepth: Field[type_ngptlw],
+           odtot: Field[type_ngptlw],
+           odcld: Field[type_ngptlw],
+           atrtot: Field[type_ngptlw],
+           atrgas: Field[type_ngptlw],
+           reflct: Field[type_ngptlw],
+           totfac: Field[type_ngptlw],
+           gasfac: Field[type_ngptlw],
+           flxfac: Field[type_ngptlw],
+           plfrac: Field[type_ngptlw],
+           blay: Field[type_ngptlw],
+           bbdgas: Field[type_ngptlw],
+           bbdtot: Field[type_ngptlw],
+           bbugas: Field[type_ngptlw],
+           bbutot: Field[type_ngptlw],
+           dplnku: Field[type_ngptlw],
+           dplnkd: Field[type_ngptlw],
+           radtotu: Field[type_ngptlw],
+           radclru: Field[type_ngptlw],
+           radtotd: Field[type_ngptlw],
+           radclrd: Field[type_ngptlw],
+           rad0: Field[type_ngptlw],
+           clfm: Field[type_ngptlw],
+           trng: Field[type_ngptlw],
+           gasu: Field[type_ngptlw],
+           itgas: Field[(np.int32, (ngptlw,))],
+           ittot: Field[(np.int32, (ngptlw,))],
+           ib: FIELD_2DINT):
+    from __externals__ import rec_6, bpade, tblint, eps, flxfac, heatfac, lhlw0
+    
+    # Downward radiative transfer loop.
+    # - Clear sky, gases contribution
+    # - Total sky, gases+clouds contribution
+    # - Cloudy layer
+    # - Total sky radiance
+    # - Clear sky radiance
+    with computation(FORWARD), interval(-2, -1):
+        for ig0 in range(ngptlw):
+            ib = NGB[0, 0][ig0]-1
+
+            # clear sky, gases contribution
+            odepth[0, 0, 0][ig0] = max(0.0, secdif[0, 0, 0][ib]*tautot[0, 0, 0][ig0])
+            if odepth[0, 0, 0][ig0] <= 0.06:
+                atrgas[0, 0, 0][ig0] = odepth[0, 0, 0][ig0] - 0.5*odepth[0, 0, 0][ig0]*odepth[0, 0, 0][ig0]
+                trng[0, 0, 0][ig0] = 1.0 - atrgas[0, 0, 0][ig0]
+                gasfac[0, 0, 0][ig0] = rec_6 * odepth[0, 0, 0][ig0]
+            else:
+                tblind[0, 0, 0][ig0] = odepth[0, 0, 0][ig0] / (bpade + odepth[0, 0, 0][ig0])
+                itgas[0, 0, 0][ig0] = tblint*tblind[0, 0, 0][ig0] + 0.5
```

Comment:
From our investigations is seemed it is not possible right now for itgas to be a local temporary.   We might want to add a TODO or Note explaining this feature limiter at the moment. If we wanted it to be able to be reduced to a scalar, we'd also want to use a different name in the other computation it shows up in. I think just a comment highlighting this issue that came up with this is enough to help future iterations remove this fully allocated field.